### PR TITLE
feat: replace system test with pack-n-play

### LINKED ISSUE #129: use pack-n-play for pack test
Replace our generated pack test with a test that uses https://github.com/google/pack-n-play. It will likely require some changes to `pack-n-play` module, at least use `npm pack` for packing and not just `tar`.

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
fix #129

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
Setup initial environment for replacing system-test, but we can definitely wait until mew feature of `pack-n-play` to make it cleaner and easier.

File: templates/typescript_gapic/system-test/install.ts.njk [hunk 1]
```diff
@@ -17,40 +17,21 @@ limitations under the License.
...
-      stdio: 'inherit',
-    });
-    await execa('node', ['--throw-deprecation', 'build/src/index.js'], {
-      cwd: `${stagingPath}/`,
-      stdio: 'inherit',
-    });
+    const options = {
+      packageDir: process.cwd(),  // path to your module.
+      sample: {
+        description: 'typescript based used can use my type definitions',
```
Comment (alexander-fenster) [inline templates/typescript_gapic/system-test/install.ts.njk]: used -> user?

File: templates/typescript_gapic/system-test/install.ts.njk [hunk 2]
```diff
@@ -17,40 +17,21 @@ limitations under the License.
...
+    await packNTest(options);  // will throw upon error.
   });
 
-  after('cleanup staging', () => {
-    if (!keep) {
-      stagingDir.removeCallback();
-    }
-  });
-});
+});
```
Comment (alexander-fenster) [inline templates/typescript_gapic/system-test/install.ts.njk]: new line

File: templates/typescript_gapic/system-test/install.ts.njk [hunk 3]
```diff
@@ -17,40 +17,21 @@ limitations under the License.
...
-    });
-    await execa('node', ['--throw-deprecation', 'build/src/index.js'], {
-      cwd: `${stagingPath}/`,
-      stdio: 'inherit',
-    });
+    const options = {
+      packageDir: process.cwd(),  // path to your module.
+      sample: {
+        description: 'typescript based used can use my type definitions',
+        ts: readFileSync('./system-test/fixtures/sample/src/index.ts').toString()
```
Comment (alexander-fenster) [inline templates/typescript_gapic/system-test/install.ts.njk]: if it's that simple, can we have a JS user version too? I think `pack-n-play` supports it.

**Comment by alexander-fenster** [comment]
@JustinBeckwith FYI - seems like @xiaozhenliu-gg5 made it work :)

File: templates/typescript_gapic/system-test/install.ts.njk (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply templates/typescript_gapic/system-test/install.ts.njk]: yes sure, turns out the `ts` or `js` field accepts `string`

File: typescript/test/testdata/keymanager/system-test/fixtures/sample/src/index.js.baseline [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import {KeyManagementServiceClient} from 'kms';
```
Comment (alexander-fenster) [inline typescript/test/testdata/keymanager/system-test/fixtures/sample/src/index.js.baseline]: In JavaScript, let's do `const {KeyManagementServiceClient} = require('kms');`

Comment (xiaozhenliu-gg5) [reply typescript/test/testdata/keymanager/system-test/fixtures/sample/src/index.js.baseline]: sure!

**Comment by xiaozhenliu-gg5** [comment]
Thanks Alex, Justin!

**Pull request merged**